### PR TITLE
Add groupId to api for data store creation in aqueduct

### PR DESCRIPTION
**Pull request by jatgarg**
## Description

Add groupId to api for data store creation in aqueduct.

### CODE REVIEW

File: packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts [hunk 1]
```diff
@@ -208,17 +208,22 @@ export class PureDataObjectFactory<
 	 * @param context - The context being used to create the runtime
 	 * (the created object will have its own new context created as well)
 	 * @param initialState - The initial state to provide to the created data store.
+	 * @param groupId - group to which this data stores belongs to. The datastores with groupId might not be loaded
```
Comment (Josmithr) [inline packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts:211]: What does it mean for the caller to not provide this? Can we document that?

Comment (jatgarg) [reply packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts:211]: sure.

**Comment by anthony-murphy** [comment]
i'd like to see an API council review of the planned usage here, as its not clear me how customer will use this.

File: packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts [hunk 2]
```diff
@@ -208,17 +208,23 @@ export class PureDataObjectFactory<
...
+	 * immediately. This groupId is also known at service side and can be used to fetch snapshot contents like
+	 * snapshot tree, blobs using this id from the storage. Snapshot for a datastore with no groupId will definitely
+	 * be included in the snapshot on load.
```
Comment (vladsud) [inline packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts:214]: I'd love to see a link from here to a readme.md that talks in more details about this feature. It's fine if it's empty for starters, but we need to put really great documentation on how that feature works

**Review by vladsud — APPROVED** [review]

**Comment by jatgarg** [comment]
> i'd like to see an API council review of the planned usage here, as its not clear me how customer will use this: https://eng.ms/docs/experiences-devices/opg/office-shared/fluid-framework/fluid-framework-internal/fluid-framework/docs/dev/resources/design-councils#api-council

Sure, I will go through the link that you mentioned.

**Comment by jatgarg** [comment]
Closing this PR based on the discussion that we are not planning to add these in aqueduct layer yet.

**Pull request closed without merging**